src/main/java/io/airlift/slice/Slices.java

```diff
@@ -91,6 +93,17 @@ public static Slice allocate(int capacity)
         return new Slice(new byte[capacity]);
     }
 
+    public static Slice allocate(Arena arena, int capacity)
+    {
+        if (capacity == 0) {
+            return EMPTY_SLICE;
+        }
+        if (capacity > MAX_ARRAY_SIZE) {
+            throw new SliceTooLargeException(format("Cannot allocate slice larger than %s bytes", MAX_ARRAY_SIZE));
+        }
+        return new Slice(arena.allocate(capacity));
+    }
```

Comment:
We should not support this for now.  I did a bunch of benchmarking and the perf is terrible once you use different MemorySegment implementations in this code.  For now, we should only support byte[].  We can change this if the virturalization problems are fixed, but I think the longer term plan is to not use Slice at all.